app/src/main/java/com/example/seoulgonggong/data/repository/DefaultWeatherRepository.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.seoulgonggong.data.repository
+
+import android.accounts.NetworkErrorException
+import com.example.seoulgonggong.data.model.ForecastResponse
+import com.example.seoulgonggong.data.service.WeatherService
+import com.example.seoulgonggong.domain.model.Weather
+import com.example.seoulgonggong.domain.model.WeatherCategory.PTY
+import com.example.seoulgonggong.domain.model.WeatherCategory.SKY
+import com.example.seoulgonggong.domain.model.WeatherCategory.TMP
+import com.example.seoulgonggong.domain.repository.WeatherRepository
+import javax.inject.Inject
+
+class DefaultWeatherRepository
+    @Inject
+    constructor(
+        private val weatherService: WeatherService,
+    ) : WeatherRepository {
+        override suspend fun getWeather(
+            serviceKey: String,
+            baseDate: String,
+            baseTime: String,
+            nx: Int,
+            ny: Int,
+        ): Weather {
+            val response = weatherService.getForecast(serviceKey, baseDate, baseTime, nx, ny)
+            if (response.isSuccessful) {
+                val forecasts = response.body()?.response?.body?.items?.item.orEmpty()
+                val forecastDateTimeMap: MutableMap<String, Weather> = sortForecastByTime(forecasts)
+
+                val list = forecastDateTimeMap.values.toMutableList()
+                list.sortWith { f1, f2 ->
+                    val f1DateTime = "${f1.forecastDate}${f1.forecastTime}"
+                    val f2DateTime = "${f2.forecastDate}${f2.forecastTime}"
+
+                    return@sortWith f1DateTime.compareTo(f2DateTime)
+                }
+                return list.first()
+            } else {
+                throw NetworkErrorException()
+            }
+        }
+
+        private fun sortForecastByTime(forecasts: List<ForecastResponse>): MutableMap<String, Weather> {
+            val forecastDateTimeMap = mutableMapOf<String, Weather>()
+            for (forecast in forecasts) {
+                if (forecastDateTimeMap["${forecast.forecastDate}/${forecast.forecastTime}"] == null) {
+                    forecastDateTimeMap["${forecast.forecastDate}/${forecast.forecastTime}"] =
+                        Weather(forecast.forecastDate, forecast.forecastTime)
+                }
```

Comment:
```suggestion
if (forecastDateTimeMap["${forecast.forecastDate}/${forecast.forecastTime}"] == null) {
                    var precipitationType: String = ""
                    var sky: String = ""
                    var temperature: Double = 0.0

                    when (forecast.category) {
                        PTY.toString() -> precipitationType = forecast.transformRainType()
                        SKY.toString() -> sky = forecast.transformSky()
                        TMP.toString() -> temperature = forecast.forecastValue.toDouble()
                        else -> {}
                    }

                    forecastDateTimeMap["${forecast.forecastDate}/${forecast.forecastTime}"] =
                        Weather(forecast.forecastDate, forecast.forecastTime,temperature, sky, precipitationType)
                }
```

대충 이런식으로 Weather 값을 한번에 넣어주는 것은 어떨까요? 도메인 모델인데 var로 언제든 밖에서 바꿀 수 있게 두는 것이 좀 불안해서 val 로 할 수 있는 방법을 생각해보았습니다. 근데 제가 구조를 정확하게 파악하지는 못하니 뽀또가 판단하시기에 더 적당한 방법으로 사용하시면 될 것 같아요!

Reply:
오! 진짜 제가 정신놓고 친 코드들 너무 잘 잡아내주시네요ㅋㅋㅋㅋㅋ 당신은 나만의 작은 아기 리뷰어 ( ͡° ͜ʖ ͡°)
추천해주신 방식처럼 변경하였슴다!